test/parser/expressions/template.ts

```diff
@@ -338,7 +338,15 @@ describe('Expressions - Template', () => {
     'test`\\18`;',
     '(`\n`)',
     '(`\r`)',
-    'new nestedNewOperatorFunction`1``2``3``array`'
+    'new nestedNewOperatorFunction`1``2``3``array`',
+    "tag()`'\\00a0'`;",
+    "(tag = () => {})`'\\00a0'`",
+    "(() => {})`'\\00a0'`",
+    "(function tag() { return () => {}; })()`'\\00a0'`",
+    "(function() { return () => {}; })()`'\\00a0'`",
+    "(function tag() {})`'\\00a0'`",
+    "(function() {})`'\\00a0'`",
+    'String.raw`{\rtf1adeflang1025ansiansicpg1252\\uc1`;'
```

Comment:
Looks like the case from [the ticket](https://github.com/surfly/it/issues/4752) is still not working.
Tested on https://meriyah.github.io/meriyah/

```
String.raw`{\rtf1\adeflang1025\ansi\ansicpg1252\uc1`;

// > [1:49]: Invalid hexadecimal escape sequence
```

Reply:
Actually none of my examples are working in this app and I believe it's because that my fix is available only in version v4.4.2. The version of `https://meriyah.github.io/meriyah/` is v4.3.4
<img width="331" alt="Screenshot 2024-04-04 at 11 31 28" src="https://github.com/surfly/meriyah/assets/6194601/e4bcc8c4-483f-4945-9f1a-45e7ed9f0e5d">

Reply:
There is a specific test for this case https://github.com/surfly/meriyah/pull/29/files#diff-9da4b71cb6cc09891f2ad28992c7cf414cf7c39d26adcc67c66069f22d4e1684R4132 and two more for other main cases.

Reply:
You're right, online version of the parser is outdated.

But the test cases are actually working in old versions, that caused the confusion:

```js
// string from the tests - parsed before the fix
String.raw`{\rtf1adeflang1025ansiansicpg1252\\uc1`;

// string from the ticket - only works after the fix
String.raw`{\rtf1\adeflang1025\ansi\ansicpg1252\uc1`;
```
Can we add second string to the tests?

Reply:
Aha, I see your confusion and it's tricky indeed! I must have added an additional '\' because otherwise a js string can't be created:
<img width="418" alt="Screenshot 2024-04-04 at 18 47 16" src="https://github.com/surfly/meriyah/assets/6194601/c2afc202-d7b8-48f8-a662-afe36f5145ef">

Reply:
Not as a regular string, but `String.raw` actually accepts it.
![image](https://github.com/surfly/meriyah/assets/11188888/afd8c992-2419-4617-bb40-ac19730e4122)